CONTRIBUTING.md

```diff
@@ -1,102 +1,181 @@
 
-# How to file an issue
+```
+// sanity.json
+"__experimental_spaces": [
+  {
+    "name": "yourDatasetName1",
+    "title": "Your Dataset Name 1",
+    "default": true,
+    "api": {
+    	"dataset": "yourDataset1"
+  	}
+  },
+  {
+    "name": "yourDatasetName2",
+    "title": "Your Dataset Name 2",
+    "api": {
+      "dataset": "yourDataset2"
+    }
+  }
+],
+"api": {
+  "projectId": "yourProjectId",
+  "dataset": "yourDataset1"
+}
+```
 
-If you find a security vulnerability, do **NOT** open an issue. Email security@sanity.io instead.
+Next, start the Test Studio by running `npm run start` in the terminal (while still in the root folder) and open it in your favorite browser to see it live.
 
-Any security issues should be submitted directly to security@sanity.io. In order to determine whether you are dealing with a security issue, ask yourself these two questions:
+```
+npm run start
+```
 
-- Can I access something that's not mine, or something I shouldn't have access to?
-- Can I disable something for other people?
+## Working on fixes or features
 
-If the answer to either of those two questions are "yes", then you're probably dealing with a security issue. Note that even if you answer "no" to both questions, you may still be dealing with a security issue, so if you're unsure, just email us at security@sanity.io.
+Please create a new branch branch for the fix or feature/enhancement you want to work on and regularly push your work to this branch. Remember to also keep your branch up to date by rebasing it regularly against the original `next` branch. To do this, add the original Sanity repository as a remote (in our example we called it `upstream`).
 
-## How to report a bug
+```
+git remote add upstream git@github.com:sanity-io/sanity.git
+git pull upstream next --rebase
+```
 
-When filing an issue, make sure to answer these six questions:
+### New features
 
-- Which versions of Sanity are you using (`sanity versions`)?
-- What operating system are you using?
-- Which versions of Node.js / npm are you running?
-- What did you do?
-- What did you expect to see?
-- What did you see instead?
+Please note that if you would like to develop a new feature for the Sanity Studio, it's unlikely that we will be able to accept your contribution. However, we highly recommend that you develop and publish your new feature as a plugin. Plugins can be [parts](https://www.sanity.io/docs/extending/parts), [custom input components](https://www.sanity.io/docs/extending/custom-input-widgets), [dashboard widgets](https://www.sanity.io/docs/dashboard/creating-your-own-widget), and [Studio themes](https://www.sanity.io/docs/guides/how-to-brand-your-studio) that you and others can use to customize and extend the functionality of the Studio.
```

Comment:
GitHub shows this paragraph really strange. I don't know if it was something we did.

Reply:
Should we also mention that if they throw us a message beforehand and discuss their "new feature" or similar in an issue, we might accept it? Discussions started from the community might be very valuable input for stuff we create anyways, and if we encourage them to start a discussion in an issue, they would also feel more welcomed.